overlord/devicestate/handlers_install.go

```diff
@@ -1378,5 +1387,12 @@ func deleteOldSaveKey(saveMntPnt string) error {
 		"factory-reset-old-fallback": true,
 	}
 
-	return secbootDeleteKeys(diskPath, toDelete)
+
+	if err := secbootDeleteKeys(diskPath, toDelete); err != nil {
+		return fmt.Errorf("cannot delete previous keys: %w", err)
+	}
+	if err := secbootRemoveOldDiskKeys(diskPath); err != nil {
```

Comment:
it a bit strange that we both call DeleteKeys and then call a helper that does a bit of that as well, at a minimum these need comments about what each does

Reply:
I have added comments before the calls to explain what both are doing.